src/CoreWCF.Primitives/src/CoreWCF/Channels/BinaryMessageEncoderFactory.cs

```diff
@@ -606,15 +601,19 @@ public override Message ReadMessage(ArraySegment<byte> buffer, BufferManager buf
                 Message message;
                 if (_messagePatterns != null)
                 {
-                    message = _messagePatterns.TryCreateMessage(buffer.Array, buffer.Offset, buffer.Count, bufferManager, messageData);
+                    int bufferLength = (int)buffer.Length;
+                    byte[] bytes = bufferManager.TakeBuffer(bufferLength);
+                    buffer.CopyTo(bytes.AsSpan(0, bufferLength));
+                    message = _messagePatterns.TryCreateMessage(bytes, 0, bufferLength, bufferManager, messageData);
                 }
```

Comment:
We should have a follow up task to convert MessagePatterns to use ReadOnlySequence

Reply:
done